### PR TITLE
Customizable whitespace-cleanup

### PR DESCRIPTION
**Pull request by nixmaniack**
- Default: cleanup whitespace on changed lines

### CODE REVIEW

**Comment by StreakyCobra** [comment]
Perfect :+1: :smile:

File: layers/+distribution/spacemacs-base/packages.el [hunk 1]
```diff
@@ -1467,6 +1468,15 @@ ARG non nil means that the editing style is `vim'."
       (spacemacs|diminish whitespace-mode " ⓦ" " w")
       (spacemacs|diminish global-whitespace-mode " Ⓦ" " W"))))
 
+(defun spacemacs-base/init-ws-butler ()
+  (use-package ws-butler
+    :defer t
+    :if (eq 'changed dotspacemacs-whitespace-cleanup)
+    :init
+    (ws-butler-global-mode 1)
```
Comment (TheBB) [inline layers/+distribution/spacemacs-base/packages.el]: If you enable the mode in `:init` there's no point to make it deferred. Just put everything under `:config` and remove the deferred.

Comment (nixmaniack) [inline layers/+distribution/spacemacs-base/packages.el]: Changed!

**Comment by brandoncc** [comment]
This is something I was looking for, thanks!

**Comment by TheBB** [comment]
Thanks! I love this one :+1:. Cherry-picked in develop. You can safely delete your branch.

**Comment by d12frosted** [comment]
Wow, I love this change as well. Didn't know that you can cleanup only changed lines. Awesome.

**Pull request closed without merging**